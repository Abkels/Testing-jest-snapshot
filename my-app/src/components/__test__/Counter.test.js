import {render, fireEvent, screen} from '@testing-library/react'
import renderer from 'react-test-renderer' //grabs the snapshot of a dom tree
import Counter from '../Counter'

// test('increment counter', ()=>{
//     // render component in the virtual dom
//     render(<Counter />)

//     // select element to interact...give Id ti the test element
//     const counter = screen.getByTestId('counter')
//     const increment = screen.getByTestId('increment')
   
//     // Interact with elements
//     fireEvent.click(increment)

//     expect(counter).toHaveTextContent('2')

// })

// describe('Counter', ()=> {
//     it('it should increment counter', ()=>{
//         render(<Counter />)

//     // select element to interact...give Id ti the test element
//     const counter = screen.getByTestId('counter')
//     const increment = screen.getByTestId('increment')
   
//     // Interact with elements
//     fireEvent.click(increment)
//     expect(counter).toHaveTextContent('2')
//     });

//     it('should decrement counter', ()=>{
//         describe('Counter', ()=> {
//             it('it should increment counter', ()=>{
//                 render(<Counter />)
        
//             // select element to interact...give Id ti the test element
//             const counter = screen.getByTestId('counter')
//             const decrement = screen.getByTestId('increment')
           
//             // Interact with elements
//             fireEvent.click(decrement)
//             expect(counter).toHaveTextContent('0')
//             })
//         })
//     })
// })

describe('Counter', () => {
    it('should increment counter', () => {
      const { getByText } = render(<Counter />);
      const incrementButton = getByText('+');
      const counterValue = getByText('1');
  
      fireEvent.click(incrementButton);
  
      expect(counterValue).toHaveTextContent('2');
    });
  
    it('should decrement counter', () => {
      const { getByText } = render(<Counter />);
      const decrementButton = getByText('-');
      const counterValue = getByText('1');
  
      fireEvent.click(decrementButton);
  
      expect(counterValue).toHaveTextContent('0');
    });
  });
  
// using snapshot -- use if you dont want your UI to change unexpectedly...Install npm i react-test-renderer
// describe('Counter Snapshot', () => {
//     it('should match DOM snapshot', () => {
//         const tree = renderer .create(<Counter />).JSON();
//         expect(tree).toMatchSnapshot();
//     })
// })

test('should match DOM snapshot', ()=> {
    const {asFragment} = render(<Counter />);
    const componentSnapshot = asFragment();
    expect(componentSnapshot).toMatchSnapshot();
});