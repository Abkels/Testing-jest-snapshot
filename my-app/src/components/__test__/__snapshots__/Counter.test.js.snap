// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match DOM snapshot 1`] = `
<DocumentFragment>
  <div
    class="counter"
  >
    <button
      data-testid="increment"
    >
      +
    </button>
    <p
      data-testid="counter"
    >
      1
    </p>
    <button
      data-testid="decrement"
    >
      -
    </button>
  </div>
</DocumentFragment>
`;
